### DISCUSSION TITLE
General question: updating python packages, RPi and android

### DISCUSSION THREAD

**Discussion by AnthonyTheKoala** [modular/modular#186] [General]
**introduction**
I am keen to learn about the new mojo. Given that mojo is a superset of python, I would like to know the following.

**importing python packages into mojo**
Given that mojo is a superset of python, will I be able to install python packages such as:
```
pip install -U tensorflow
pip install -U opencv
pip install -U matplotlib 
```
And mojo will recognize that installation

**installing and using mojo on the RPi**
Which RPi model will be suitable for installing and using mojo

**Installing and using mojo on android**
I am also interested in writing apps for android. But the python implementation in android is limited to version 3.8.
I would like to see an android version of mojo so I can write apps.

Thank you
Anthony of Sydney

**Comment by iamtimdavis** [comment]
1. Yes when we enable local execution (see https://github.com/modularml/mojo/discussions/158#discussioncomment-5930310). For now the playground has a tighter security policy but we encourage you to suggest packages - see https://github.com/modularml/mojo/discussions/173
2. This is something that will come down the line when we support local execution and relevant RPi hardware (if it has a LLVM backend it will be easier, otherwise it will be more challenging).
3. Similarly to 2

**Reply by AnthonyTheKoala** [reply]
Thank you. I have replied to the question of the packages I would like to see added.
_Two more questions please_
Will the compiled code from mojo be executable or interpreted?

In a similar vein to my last two questions, will one be able to write in mojo to embedded devices. Eg you have a washing machine or "widget" or "gadget" with code loaded in a devices ROM. Anything to write code in say an MS OS but download the program in a "widget's" ROM?

Thank you
Anthony of Sydney

**Comment by AnthonyTheKoala** [comment]
Dear Mr Davis,
Given that Mojo is online for all the major OS such as Linux, MS Win, what about android?

This is exciting.

Thank you
Anthonyn Pleticos, Sydney

________________________________
